src/views/Requests.svelte

```diff
@@ -21,24 +20,11 @@ let showAsList = false
 
 $: queryStringData = qs.parse($querystring)
 $: requestFilter = populateRequestFilterFrom(queryStringData, $me, $events)
-$: showAsList = queryStringData.hasOwnProperty('list')
-
-function viewAsGrid() {
-  updateQueryString({ list: false })
-
-  viewedRequestsAsGrid()
-}
-
-function viewAsList() {
-  updateQueryString({ list: null })
-
-  viewedRequestsAsList()
-}
 </script>
 
 <FilteredDisplay title="Requests" filter={requestFilter} items={$requests}>
   <div slot="toggles">
-    <GridListToggle on:list={viewAsList} on:grid={viewAsGrid} {showAsList} buttonCssClass="my-1 mx-0" />
+    <GridListToggle on:list={() => showAsList = true} on:grid={() => showAsList = false} class="my-1 mx-0" />
```

Comment:
I'm a little concerned that we're coupling the GridListToggle too tightly to requests. Initially, it was intended merely as a toggle, reusable wherever is convenient. Now that it's internally storing the user's preference for whether to use the grid or list view, that separation of "Tell me what state to show, and I'll tell you when the user selects 'list' or 'grid'" now seems a little muddied. 

A related question is about this `showAsList` here in Requests.svelte: it's being set, but I don't see anywhere that it's used any more... or am I simply overlooking it?

All that said, we could very well make this Request-specific if we so choose. Feel free to ping me and we can discuss (at your convenience).

Reply:
https://github.com/silinternational/wecarry-ui/pull/152/commits/a176bb50c33a09f5e25a2edeb96cb1246158fc33